src/components/SearchPlayers.js

```diff
@@ -0,0 +1,94 @@
+//@flow strict
+
+import * as React from 'react';
+
+type Props = {|
+	playerList: Array<string>,
+|};
+
+type State = {|
+	searchName: string,
+	searchList: Array<string>,
+	displayList: React.Node
+|};
+
+// Search Players Component. Takes an array of all players.
+//
+// Prop Names:
+// playerList (Array of all users)
+// 
+// EXAMPLE PROP USAGE = <SearchPlayers playerList={ ["jeffery", "jeff", "john", "illeatbutt", "childeatbigtime", "childconsumer"] }/>
+//
+// Needs to be updated to have a challenge player button.
+class SearchPlayers extends React.Component<Props, State> {
+
+	constructor(props: Props) {
+		super(props);
+
+		this.state = {
+			searchName: "",
+			searchList: [""],
+			displayList: <div className="searchPlayersList"></div>
+		}
+	}
+
+	// When typing in the searchbar, update searchName.
+	handleChange = (e: string) => {
+		this.setState({
+			searchName: e
+		});
+	}
+
+	// When a user clicks search, handleSearch will activate.
+	handleSearch = () => {
+
+		const playerList: Array<string> = this.props.playerList;
+		const searchName: string = this.state.searchName;
+
+		// Create a filtered list showing user's desired results.
+		let list: Array<string> = [""];
+
+		let i: number = 0;
+		for(i; i < playerList.length; i++) {
+			if(playerList[i].includes(searchName)) {
+				list.push(playerList[i]);
+			}
+		}
+
+		// If no changes were made to list, then the search was empty.
+		if(list.length === 1) {
+			this.setState({ displayList:
+				<div className="searchPlayersList">
+					<h6>No users found.</h6>
+				</div>
+			});
+		} else {
+			// Map the users that were found.
+			this.setState({ displayList:
+				<div className="searchPlayersList">
+					<ul>
+						{list.map((name, index) => <button key={index} type="button" className="clearbtn searchPlayerListItem" onClick={this.challengePlayer}>{name}</button>)}
+					</ul>
+				</div>
+			});
+		}
+	}
+
+	// Challenge the player chosen.
+	challengePlayer(): void {
+		console.log("empty");
+	}
+
+	render(): React.Node {
+		return (
+			<div className="searchPlayers">
+				<h6>Click the Player's Name you wish to Challenge</h6>
+				<input type="text" className="inputText" onChange={(e) => this.handleChange(e.target.value)} placeholder="Search Players" />

```

Comment:
I'm pretty sure you are double-wrapping this and you don't need to be.

Keep this as {(e)=>this.handleChange(...)}

On the next line chance onClick to be {()=>this.handleSearch()}

Then make handleSearch and handleChange actual methods. I.E. handleChange(e): void {...}